src/cdk/layout/breakpoints-observer.ts

```diff
@@ -0,0 +1,100 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {Injectable, NgZone, OnDestroy} from '@angular/core';
+import {MediaMatcher} from './media-matcher';
+import {Observable} from 'rxjs/Observable';
+import {Subject} from 'rxjs/Subject';
+import {RxChain, map, startWith, takeUntil} from '@angular/cdk/rxjs';
+import {coerceArray} from '@angular/cdk/coercion';
+import {combineLatest} from 'rxjs/observable/combineLatest';
+import {fromEventPattern} from 'rxjs/observable/fromEventPattern';
+
+/** The current state of a layout breakpoint. */
+export interface BreakpointState {
+  query: string;
+  matches: boolean;
+}
+
+interface Query {
+  observable: Observable<BreakpointState>;
+  mql: MediaQueryList;
+}
+
+@Injectable()
+export class BreakpointObserver implements OnDestroy {
+  /**  A map of all media queries currently being listened for. */
+  private _queries: Map<string, Query> = new Map();
+  /** A subject for all other observables to takeUntil based on. */
+  private _activeSubject: Subject<{}> = new Subject();
+
+  constructor(private mediaMatcher: MediaMatcher, private zone: NgZone) {}
+
+  /** Completes the active subject, signalling to all other observables to complete. */
+  ngOnDestroy() {
+    this._activeSubject.next();
+    this._activeSubject.complete();
+  }
+
+  /** Whether the query currently is matched. */
+  isMatched(value: string | string[]): boolean {
+    let queries = coerceArray(value);
+    return queries.some((mediaQuery: string) => {
```

Comment:
The `: string` shouldn't be necessary since it is inferred from the array. It's also acceptable to use a single-character variable for concise lambdas like this:
```ts
return queries.some(q => this._registerQuery(q).mql.matches);
```